tests/tests/class-linking.php

```diff
@@ -0,0 +1,178 @@
+<?php
+namespace HM\Tachyon\Tests;
+
+use Tachyon;
+use WP_UnitTestCase;
+
+/**
+ * Test rescheduling an event is successful.
+ *
+ * @ticket 64
+ */
+class Tests_Linking extends WP_UnitTestCase {
+
+	/**
+	 * @var int[] Attachment IDs
+	 */
+	static $attachment_ids;
+
+	/**
+	 * Set up attachments and posts require for testing.
+	 *
+	 * tachyon.jpg: 1280x719
+	 * tachyon-large.jpg: 5312x2988
+	 * Photo by Digital Buggu from Pexels
+	 * @link https://www.pexels.com/photo/0-7-rpm-171195/
+	 */
+	static public function wpSetUpBeforeClass( $factory ) {
+		/**
+		 * Filter the uploads directory to avoid file name clashes in
+		 * subsequent tests.
+		 */
+		add_filter( 'upload_dir', function( $upload_dir ) {
+			$dir = strtolower( str_replace( [ '_', '/', '\\' ], '-', __CLASS__ ) );
+			$upload_dir['basedir'] = preg_replace( '#/uploads(/|$)#', "/uploads/{$dir}\$1", $upload_dir['basedir'] );
+			$upload_dir['path'] = preg_replace( '#/uploads(/|$)#', "/uploads/{$dir}\$1", $upload_dir['path'] );
+			return $upload_dir;
+		} );
+
+		self::$attachment_ids['tachyon'] = $factory->attachment->create_upload_object(
+			realpath( __DIR__ . '/../data/tachyon.jpg')
+		);
+	}
+
+	/**
+	 * Runs the routine after all tests have been run.
+	 */
+	public static function tearDownAfterClass() {
+		array_map( function( $post_id ) {
+			wp_delete_attachment( $post_id, true );
```

Comment:
Hmm do we need to delete? I thought the WP unit tests automatically rolled back the transaction between test?

Reply:
The test suite reverts the post table but doesn’t delete the related files. This allows local re-runs of the suite without the need to delete the files.

I’ll take a look at an unlink loop instead

Reply:
Switched tear down to delete files and leave the suite to deal with attachments in 907d1ecc53c359d889166085063782142d358048